pkg/build/apis/build/validation/validation.go

```diff
@@ -679,17 +679,16 @@ func ValidateBuildLogOptions(opts *buildapi.BuildLogOptions) field.ErrorList {
 	return allErrs
 }
 
-var cIdentifierErrorMsg string = `must be a C identifier (mixed-case letters, numbers, and underscores): e.g. "my_name" or "MyName"`
-
 func ValidateStrategyEnv(vars []kapi.EnvVar, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
 	for i, ev := range vars {
 		idxPath := fldPath.Index(i)
 		if len(ev.Name) == 0 {
 			allErrs = append(allErrs, field.Required(idxPath.Child("name"), ""))
-		} else if len(kvalidation.IsCIdentifier(ev.Name)) > 0 {
-			allErrs = append(allErrs, field.Invalid(idxPath.Child("name"), ev.Name, cIdentifierErrorMsg))
+		}
+		if errs := kvalidation.IsEnvVarName(ev.Name); len(errs) != 0 {
+			allErrs = append(allErrs, field.Invalid(idxPath.Child("name"), ev.Name, strings.Join(errs, "; ")))
```

Comment:
why not else-if?  you're just going to give someone two errors on an empty value now.

Reply:
kvalidation.IsEnvVarName would have to be run a second time to get the error message inside of the else.

Reply:
```
if len(env.Name) ==0 {
   allErrs= 
} else {
  if errs:=kvalidation.IsEnvVarName(); len(errs)!=0 {
    allErrs= append
  }
}
```

Reply:
also 
```
	if a:=false; a {
	  fmt.Println(a)
	} else if b:=true; b {
	  fmt.Println(b)
	}
	
```

is valid syntax.

Reply:
so you don't even have to do what i wrote